doc/drop_counters/drop_counters_HLD.md

```diff
@@ -0,0 +1,353 @@
+    - [3.3 State DB](#32-state-db)
+        - [3.3.1 SAI APIs](#321-sai-apis)
+    - [3.4 Counters DB](#34-counters-db)
+    - [3.5 SWSS](#35-swss)
+        - [3.5.1 SAI APIs](#351-sai-apis)
+    - [3.6 syncd](#34-syncd)
+* [4 Flows](#4-flows)
+    - [4.1 General Flow](#41-general-flow)
+* [5 Warm Reboot Support](#5-warm-reboot-support)
+* [6 Unit Tests](#6-unit-tests)
+* [7 Open Questions](#7-open-questions)
+
+
+
+# List of Tables
+* [Table 1: Abbreviations](#definitionsabbreviation)
+
+# List of Figures
+* [Figure 1: General Flow](#41-general-flow)
+
+# Revision
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 07/30/19 | Danny Allen | Initial version    |
+| 0.2 | 09/03/19 | Danny Allen | Review updates     |
+
+# About this Manual
+This document provides an overview of the implementation of configurable packet drop counters in SONiC.
+
+# Scope
+This document describes the high level design of the configurable drop counter feature.
+
+# Definitions/Abbreviation
+| Abbreviation | Description     |
+|--------------|-----------------|
+| RX           | Receive/ingress |
+| TX           | Transmit/egress |
+
+# 1 Overview
+The main goal of this feature is to provide better packet drop visibility in SONiC by providing a mechanism to count and classify packet drops that occur due to different reasons. 
+
+The other goal of this feature is for users to be able to track the types of drop reasons that are important for their scenario. Because different users have different priorities, and because priorities change over time, it is important for this feature to be easily configurable.
+
+We will accomplish both goals by adding support for SAI debug counters to SONiC. 
+* Support for creating and configuring port-level and switch-level debug counters will be added to orchagent and syncd. 
+* A CLI tool will be provided for users to manage and configure their own drop counters
+
+# 2 Requirements
+
+## 2.1 Functional Requirements
+1. CONFIG_DB can be configured to create debug counters
+2. STATE_DB can be queried for debug counter capabilities
+3. Users can access drop counter information via a CLI tool
+    1. Users can see what capabilities are available to them
+        1. Types of counters (i.e. port-level and/or switch-level)
+        2. Number of counters
+        3. Supported drop reasons
+    2. Users can see what types of drops each configured counter contains
+    3. Users can add and remove drop reasons from each counter
+    4. Users can read the current value of each counter
+    5. Users can assign aliases to counters
+    6. Users can clear counters
+
+## 2.2 Configuration and Management Requirements
+Configuration of the drop counters can be done via:
+* config_db.json 
+* minigraph.xml
+* CLI
+
+## 2.3 Scalability Requirements
+Users must be able to use all counters and drop reasons provided by the underlying hardware.
```

Comment:
ex: when we delete drop counters will not delete the other hanrdware counters? hope these counters are not overlap with other hardware counters?

Reply:
Correct. These counters are separate from the existing hardware counters and their behavior (e.g. deleting, clearing, etc.) will not affect the existing counters. I've added a note.